data-prepper-plugins/saas-source-plugins/saas-crawler/src/main/java/org/opensearch/dataprepper/plugins/source/saas_crawler/base/SaasClient.java

```diff
@@ -0,0 +1,43 @@
+package org.opensearch.dataprepper.plugins.source.saas_crawler.base;
+
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.plugins.source.saas_crawler.coordination.state.SaasWorkerProgressState;
+import org.opensearch.dataprepper.plugins.source.saas_crawler.model.ItemInfo;
+
+import java.util.Iterator;
+
+/**
+ * Interface for saas client. This interface can be implemented by different saas clients.
+ * For example, Jira, Salesforce, ServiceNow, etc.
+ */
+public interface SaasClient {
```

Comment:
Perhaps this can be renamed to `CrawlerClient`.

Reply:
Renamed